view/src/components/surplus.js

```diff
@@ -0,0 +1,551 @@
+import React, { Component } from "react";
+import CardSkeletons from "../extras/skeleton";
+import SurplusStepper from "../extras/surplusStepper";
+
+import withStyles from "@material-ui/core/styles/withStyles";
+import Typography from "@material-ui/core/Typography";
+import Button from "@material-ui/core/Button";
+import Dialog from "@material-ui/core/Dialog";
+import AppBar from "@material-ui/core/AppBar";
+import Toolbar from "@material-ui/core/Toolbar";
+import IconButton from "@material-ui/core/IconButton";
+import CloseIcon from "@material-ui/icons/Close";
+import Slide from "@material-ui/core/Slide";
+import Grid from "@material-ui/core/Grid";
+import Card from "@material-ui/core/Card";
+import Container from "@material-ui/core/Container";
+import CardActions from "@material-ui/core/CardActions";
+import CardContent from "@material-ui/core/CardContent";
+import MuiDialogTitle from "@material-ui/core/DialogTitle";
+import MuiDialogContent from "@material-ui/core/DialogContent";
+import InputBase from "@material-ui/core/InputBase";
+import SearchIcon from "@material-ui/icons/Search";
+import Box from "@material-ui/core/Box";
+import Fab from "@material-ui/core/Fab";
+import AddIcon from "@material-ui/icons/Add";
+
+import axios from "axios";
+import dayjs from "dayjs";
+import relativeTime from "dayjs/plugin/relativeTime";
+import { authMiddleWare } from "../util/auth";
+
+const styles = (theme) => ({
+  content: {
```

Comment:
Same comments as other PR with regard to CSS values being strings and having units, and the width: 98% thing.  I imagine this probably got copied from the original template and should have been caught then, sorry.

(Part of why I wish I knew React better is that for all I know, React automatically converts numeric values here to strings with a unit of "px"...  it's certainly never struck me as being terribly interested in type safety.  If that's the case, then feel free to follow whatever the best practices seem to be here, just try to be consistent wherever possible.)

Reply:
I decided to put all the margins in "px" so that it is consistent